caddy/tasks/main.yml

```diff
@@ -41,16 +41,18 @@
     mode: 0755
 
 - name: determine installed caddy version
-  shell: "caddy version | awk '{ print $1 }' || true"
+  shell: caddy version | awk '{ gsub(/v/,""); print $1 }' || true
   register: caddy_version_output
 
 - name: set caddy_installed_version fact
   set_fact: caddy_installed_version="{{caddy_version_output.stdout}}"
 
 - name: install caddy
   shell: |
+    set -e
     cd `mktemp -d`
-    curl -fsSL -o caddy 'https://caddyserver.com/api/download?os=linux&arch=amd64'
+    curl -fsSL -o caddy.tar.gz 'https://github.com/caddyserver/caddy/releases/download/v{{caddy_version}}/caddy_{{caddy_version}}_linux_amd64.tar.gz'
+    tar -xf caddy.tar.gz
     install -m 0755 caddy /usr/local/bin/caddy
     setcap 'cap_net_bind_service=+ep' /usr/local/bin/caddy
   when: caddy_installed_version != caddy_version
```

Comment:
I'd prefer to make the check work for both `1.2.3` and `v1.2.3` by also stripping leading `v`'s from `caddy_version` (i.e. the user provided variable). As is, the role would re-install caddy on every run if the user would set `caddy_version` to for example `v2.7.6` (which is the output of `caddy version`, so the old default seems more in line with what a user would expect). What do you think?

Reply:
I agree, I implemented my suggestion on how to do it and tested with a leading 'v' in the variable and without.
Let me know how this looks to you :)